app/utils/build-url-with-prefix-map.js

```diff
@@ -0,0 +1,39 @@
+import Ember from 'ember';
+
+export default function buildURLWithPrefixMap(prefixPropertyMapping){
+  return function buildURL(type, id, record) {
+    var url = [],
+        host = Ember.get(this, 'host'),
+        prefix = this.urlPrefix();
```

Comment:
style-wise, multiple `var` statements are easier to debug (you can drop a breakpoint on any of them).